kotlin-analysers/src/main/kotlin/org/jetbrains/research/lupa/kotlinAnalysis/stdlib/analysis/StdlibInterfacesAnalyzer.kt

```diff
@@ -0,0 +1,77 @@
+package org.jetbrains.research.lupa.kotlinAnalysis.stdlib.analysis
+
+import org.jetbrains.kotlin.backend.jvm.ir.psiElement
+import org.jetbrains.kotlin.descriptors.ClassDescriptor
+import org.jetbrains.kotlin.descriptors.FunctionDescriptor
+import org.jetbrains.kotlin.idea.caches.resolve.resolveToDescriptorIfAny
+import org.jetbrains.kotlin.psi.KtClass
+import org.jetbrains.kotlin.resolve.descriptorUtil.fqNameOrNull
+import org.jetbrains.kotlin.resolve.descriptorUtil.fqNameSafe
+import org.jetbrains.kotlin.resolve.descriptorUtil.getAllSuperClassifiers
+import org.jetbrains.kotlin.resolve.lazy.descriptors.LazyClassDescriptor
+import org.jetbrains.research.lupa.kotlinAnalysis.PsiAnalyzer
+
+data class StdlibInterfacesAnalysisResult(
+    val functionName: String,
+    val interfaceName: String,
+    val baseInterfaces: List<String>,
+)
+
+object StdlibInterfacesAnalyzer : PsiAnalyzer<KtClass, List<StdlibInterfacesAnalysisResult>?> {
+    private val baseInterfaces = listOf(
+        "kotlin.coroutines.ContinuationInterceptor",
+        "kotlin.coroutines.CoroutineContext",
+        "kotlin.coroutines.CoroutineContext.Element",
+        "kotlin.ranges.ClosedRange",
+        "kotlin.ranges.OpenEndRange",
+        "kotlin.ranges.ClosedFloatingPointRange",
+        "kotlin.text.MatchResult",
+        "kotlin.time.TimeMark",
+        "kotlin.time.ComparableTimeMark",
+    )
+
+    private fun KtClass.getSuperTypes(): List<ClassDescriptor> =
+        this.resolveToDescriptorIfAny()?.getAllSuperClassifiers()?.filterIsInstance<ClassDescriptor>()?.filter {
+            it.fqNameOrNull().toString() in baseInterfaces
+        }?.toList() ?: error("Can not resolve descriptor for class ${this.name}")
```

Comment:
This object doesn't use a list of descriptors anywhere, but you need to know if KtClass has descriptors or not. So it would probably be better to call this function `hasSuperTypes` and make it return Boolean:
```kotlin
    private fun KtClass.hasSuperTypes(): Boolean =
        this.resolveToDescriptorIfAny()?.getAllSuperClassifiers()?.filterIsInstance<ClassDescriptor>()?.filter {
            it.fqNameOrNull().toString() in baseInterfaces
        }?.toList()?.isEmpty() ?: error("Can not resolve descriptor for class ${this.name}")
```

Reply:
But if you need this list later, then let's leave it as it is :)

Reply:
yep, you are right, I planned to use these supertypes, but next found another solution